docs/architecture.md

```diff
@@ -0,0 +1,188 @@
+parentheses. To enhance readability, the suffix "process" has been omitted from the process descriptions below.
+
+1. (HTTPS) *NKG* reads the *Kubernetes API* to get the latest versions of the resources in the cluster and writes to the
+API to update the handled resources' statuses and emit events.
+2. (File I/O) *NKG* generates NGINX *configuration* based on the cluster resources and writes them as `.conf` files to
+the mounted `nginx` volume, located at `/etc/nginx`. It also writes *TLS certificates* and *keys*
+from [TLS Secrets][secrets] referenced in the accepted Gateway resource to the volume at the
+path `/etc/nginx/secrets`.
+3. (File I/O) *NKG* writes logs to its *stdout* and *stderr*, which are collected by the container runtime.
+4. (Signal) To reload NGINX, *NKG* sends the [reload signal][reload] to the **NGINX master**.
+5. (File I/O) The *NGINX master* reads *configuration files*  and the *TLS cert and keys* referenced in the
+configuration when it starts or during a reload. These files, certificates, and keys are stored in the `nginx` volume
+that is mounted to both the `nginx-gateway` and `nginx` containers.
+6. (File I/O): The *NGINX master* writes to the auxiliary Unix sockets folder, which is mounted to the `nginx`
+container as the `var-lib-nginx` volume. The mounted path for this volume is `/var/lib/nginx`.
+7. (File I/O) The *NGINX master* sends logs to its *stdout* and *stderr*, which are collected by the container runtime.
+8. (File I/O): The *NGINX master* reads the NJS modules referenced in the configuration when it starts or during a
+reload. NJS modules are stored in the `njs-modules` volume that is mounted to the `nginx` container.
+9. (File I/O) An *NGINX worker* writes logs to its *stdout* and *stderr*, which are collected by the container runtime.
+10. (File I/O): The *NGINX master* reads the `nginx.conf` file from the mounted `nginx-conf` volume.
+This [file][conf-file] contains the global and http configuration settings for NGINX.
+11. (Signal) The *NGINX master* controls the [lifecycle of *NGINX workers*][lifecycle] it creates workers with the new
+configuration and shutdowns workers with the old configuration.
+12. (HTTP,HTTPS) A *client* sends traffic to and receives traffic from any of the *NGINX workers* on ports 80 and 443.
+13. (HTTP,HTTPS) An *NGINX worker* sends traffic to and receives traffic from the *backends*.
+
+[controller]: https://kubernetes.io/docs/concepts/architecture/controller/
+
+[runtime]: https://github.com/kubernetes-sigs/controller-runtime
+
+[secrets]: https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets
+
+[reload]: https://nginx.org/en/docs/control.html
+
+[lifecycle]: https://nginx.org/en/docs/control.html#reconfiguration
+
+[conf-file]: https://github.com/nginxinc/nginx-kubernetes-gateway/blob/main/deploy/manifests/nginx-conf.yaml
+
+[share]: https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/
+
+## Design Principles
```

Comment:
I wonder if it make sense to put those into a separate doc, so that they are move visible. And what if we also link to it from the main README, like  below:

NGINX Kubernetes Gateway is an open-source project that provides an implementation of the [Gateway API](https://gateway-api.sigs.k8s.io/) using [NGINX](https://nginx.org/) as the data plane. The goal of this project is to implement the core Gateway APIs -- Gateway, GatewayClass, HTTPRoute, TCPRoute, TLSRoute, and UDPRoute -- to configure an HTTP or TCP/UDP load balancer, reverse-proxy, or API gateway for applications running on Kubernetes. NGINX Kubernetes Gateway is currently under development and supports a subset of the Gateway API.

For a list of supported Gateway API resources and features, see the [Gateway API Compatibility](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/main/docs/gateway-api-compatibility.md) doc.

Warning: This project is actively in development (beta feature state) and should not be deployed in a production environment. All APIs, SDKs, designs, and packages are subject to change.

Learn about our [design principles](#).